### DISCUSSION TITLE
eglot 1.8. mode-line says "eglot:build" - what does "build" mean?

### DISCUSSION THREAD

**Discussion by stephe-ada-guru** [joaotavora/eglot#1011] [General]
I found eglot--mode-line-format in eglot.el, but it doesn't have the string "build" anywhere near it - what puts the "build" there?

Perhaps the meaning of the mode-line should be documented somewhere?

**Comment by joaotavora** [comment]
Probably `build` is the name of the project you're working on as devined by `project-current`

**Reply by stephe-ada-guru** [reply]
Ah; that hint let me understand the mode-line code better. 

It's the name of default-directory; project doesn't have the concept of "name" (although I've requested that be added). Not very useful for my use cases; I always put the main project file in a "build" directory.

The mode-line is showing eglot--project-nickname, which is set by eglot-connect as:

         (nickname (file-name-base (directory-file-name default-directory)))

Maybe I can use this use-case to justify adding "project-name".

**Reply by joaotavora** [reply]
> Maybe I can use this use-case to justify adding "project-name".

Maybe, though I'd say the idea holds on its own even without this use case.

On the other hand, I personally find it odd to place the project file in a `build` directory.  I use `build` directories for the trasient, non-checked in artifacts that are the product of building my program.  My project file (which is almost always a `.git` directory) lives in a directory that is named after the project.  But this is just a personal opinion.
